src/record/list.ts

```diff
@@ -212,8 +212,13 @@ export class List extends Emitter {
 
         const listenCallback = this.wrappedFunctions.get(parameters.callback)
         parameters.callback = listenCallback as (data: any) => void
-        this.record.unsubscribe(parameters)
         this.wrappedFunctions.delete(parameters.callback)
+        this.subscriptions = this.subscriptions.filter((subscription: any) => {
+          if (parameters.callback && parameters.callback !== subscription.callback) return true
```

Comment:
tiny syntax sugar, but you can just do

this.subscriptions.filter((subscription: any) => parameters.callback !== subscription.callback)

I'm assuming callback always exists, since otherwise its broke

Reply:
Nope callback doesn't always exist. You can do `list.unsubscribe(callback)` to cancel a specif callback or `list.unsubscribe()` to get rid of all subscriptions, that's why the extra step. Because otherwise `undefined !== function` is always true